lib/AST/NameLookup.cpp

```diff
@@ -1295,8 +1272,15 @@ bool AbstractStorageDecl::isSetterAccessibleFrom(const DeclContext *DC) const {
   // accessibility, it is not set.
   if (hasStorage() && !isSettable(nullptr))
     return true;
-  
-  return checkAccessibility(DC, getDeclContext(), getSetterAccessibility());
+
+  auto setterAccess = getFormalSetterAccess(DC);
+  if (!DC)
+    return setterAccess >= Accessibility::Public;
+
+  auto scope = getDeclContext()->getAccessScope(DC, setterAccess,
+    /* useNominalTypeAccessibility */ false);
```

Comment:
This is false for the scenario of extending public protocols with private subtypes. The test that caught the behavior is `test/SILGen/witness_accessibility.swift`. 

In this test, an extension on a private protocol implements a default requirement for a parent public protocol. The current access scope lookup would restrict the visibility of the extension method `R.publicRequirement` to `fileprivate` since the type of `R` is `fileprivate`. But with this level of visibility `struct S: R` doesn't conform to protocol `P`.

Now with all that said, I'm realizing that you might just be referring to why is this false just in the case of the setter, and that's a good catch -- I don't think there's a need for that. I'll add a test to confirm.